core/src/main/java/com/linecorp/armeria/client/Http2ResponseDecoder.java

```diff
@@ -293,15 +307,13 @@ public void onPriorityRead(ChannelHandlerContext ctx, int streamId, int streamDe
 
     @Override
     public void onPingRead(ChannelHandlerContext ctx, long data) {
-        if (keepAliveHandler != null) {
-            keepAliveHandler.onPing();
-        }
+        keepAliveHandler.onPing();
     }
 
     @Override
     public void onPingAckRead(ChannelHandlerContext ctx, long data) {
-        if (keepAliveHandler != null) {
-            keepAliveHandler.onPingAck(data);
+        if (keepAliveHandler instanceof Http2ClientKeepAliveHandler) {
```

Comment:
Do you think it makes sense to add a method like `KeepAliveHandler.isClient()` to remove this kind of `instanceof` check? We could pull up `onPingAck()` as well and do nothing (or raise an exception) when called unnecessarily.

Reply:
Let me use `KeepAliveHandler.isHttp2()` because `onPickAck()` could be called both server and client.